File: 백준/Platinum/11871. 님 게임 홀짝/님 게임 홀짝.py
def ip(): return int(input())
def sp(): return str(input().rstrip())
 
def mip(): return map(int, input().split())
def msp(): return map(str, input().split().rstrip())
 
def lmip(): return list(map(int, input().split()))
def lmsp(): return list(map(str, input().split().rstrip()))
 
n = ip()
p = lmip()
ans = 0
for i in p:
    ans ^= ((i + 1) // 2 if i & 1 else (i - 2) // 2)
print("koosaga" if ans else "cubelover")

File: 백준/Platinum/11871. 님 게임 홀짝/님 게임 홀짝.cc
#include <iostream>
using namespace std;

int main(){
    int n, p, result = 0;
    cin >> n;

    for(int i=0; i<n; i++){
        cin >> p;
        if(p%2 == 0) result ^= (p-2)/2;
        else result ^= (p+1)/2;
    }

    if(result == 0) cout << "cubelover\n";
    else cout << "koosaga\n";

    return 0;
}